### PR TITLE
Restructure secure settings docs and minor additions

### LINKED ISSUE #5425: Improve the documentation around Elasticsearch secure settings
The current documentation is at https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-es-secure-settings.html

It can be very confusing to users, the `key`, `path` projection mechanism is not something users need to understand for basic usage but it is presented in one continuous paragraph with the basic use of the feature. 

We could improve this as follows:

* have three sections maybe (titles tbd): basic usage, projection/subset usage, examples
* most users will learn all they need in the basic section and can ignore the complicated projection feature
* examples: we already link to a GCS example for snapshots, we could highlight this a bit more and add another example inline maybe for AWS S3 .

### PR DESCRIPTION
**Pull request by pebrc**
Fixes #5425

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/elasticsearch/es-secure-settings.asciidoc [hunk 1]
```diff
@@ -9,7 +9,9 @@ endif::[]
 = Secure settings
 
 You can specify link:https://www.elastic.co/guide/en/elasticsearch/reference/current/secure-settings.html[secure settings] with Kubernetes secrets.
-The secrets should contain a key-value pair for each secure setting you want to add. ECK automatically injects these settings into the keystore on each Elasticsearch node before it starts Elasticsearch.
+The secrets should contain a key-value pair for each secure setting you want to add. ECK automatically injects these settings into the keystore on each Elasticsearch node before it starts Elasticsearch. The ECK operator keeps watching the secrets for changes and will update the Elasticsearch keystore when it detects a change.
```
Comment (naemono) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/es-secure-settings.asciidoc:12]:
```suggestion
The secrets should contain a key-value pair for each secure setting you want to add. ECK automatically injects these settings into the keystore on each Elasticsearch node before it starts Elasticsearch. The ECK operator continues to watch the secrets for changes and will update the Elasticsearch keystore when it detects a change.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/es-secure-settings.asciidoc [hunk 2]
```diff
@@ -70,4 +74,7 @@ data:
   gcs_client_2: RWxhc3RpYyBDbG91ZCBvbiBLOHMgKEVDSykgLSBHQ1MgY2xpZW50IDIK
 ----
 
-Check <<{p}-snapshots,How to create automated snapshots>> for an example use case.
+
+== More Examples
```
Comment (alaudazzi) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/es-secure-settings.asciidoc:78]:
```suggestion
== More examples
```

**Review by alaudazzi — APPROVED** [review]
A super minor fix, as we don't use camel case in titles :-)
LGTM

File: docs/orchestrating-elastic-stack-applications/elasticsearch/es-secure-settings.asciidoc [hunk 3]
```diff
@@ -70,4 +74,7 @@ data:
...
+
+Check <<{p}-snapshots,How to create automated snapshots>> for an example use case that illustrates how secure settings can be used to set up automated Elasticsearch snapshots to an GCS storage bucket.
```
Comment (thbkrkr) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/es-secure-settings.asciidoc:80]:
```suggestion
Check <<{p}-snapshots,How to create automated snapshots>> for an example use case that illustrates how secure settings can be used to set up automated Elasticsearch snapshots to a GCS storage bucket.
```

**Pull request merged**